libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationLocationProvider.java

```diff
@@ -0,0 +1,38 @@
+package com.mapbox.services.android.navigation.v5.navigation;
+
+import android.location.Location;
+
+import com.mapbox.android.core.location.LocationEngine;
+import com.mapbox.api.directions.v5.models.DirectionsRoute;
+import com.mapbox.services.android.navigation.v5.utils.RouteUtils;
+
+class NavigationLocationProvider {
```

Comment:
`NavigationLocationProvider` is called "provider" but doesn’t have any method that returns an object. That's confusing IMO.

Reply:
Changed to `NavigationLocationEngineUpdater`